src/app/services/local-storage.service.ts

```diff
@@ -0,0 +1,29 @@
+export class StorageService<T> {
+  private storageKeyPrefix: string;
+
+  constructor(storageKeyPrefix: string) {
+    this.storageKeyPrefix = storageKeyPrefix;
+  }
+
+  private getStorageKey(key: string): string {
+    return `${this.storageKeyPrefix}_${key}`;
+  }
+
+  public saveData<K extends keyof T>(key: K, data: T[K]): void {
+    const storageKey = this.getStorageKey(key.toString());
+    localStorage.setItem(storageKey, JSON.stringify(data));
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public getData<K extends keyof T>(key: K): any | null {
```

Comment:
```any``` can be replaced with ```T[K] | null``` This way, the return type will be consistent with the type of data being stored.
```suggestion
  public getData<K extends keyof T>(key: K): T[K] | null {
```